File: src/main/java/org/lucidfox/questfiller/model/quest/ReputationGain.java
package org.lucidfox.questfiller.model.quest;

public final class ReputationGain {
	private final String faction;
	private final String canonicalName;
	private final int gain;
	
	public ReputationGain(final String faction, final String canonicalName, final int gain) {
		this.faction = faction;
		this.canonicalName = canonicalName;
		this.gain = gain;
	}
	
	private String getLink() {
		if (canonicalName == null) {
			return String.format("[[%s]]", faction);
		} else {
			return String.format("[[%s|%s]]", canonicalName, faction);
		}
	}
	
	public String getFullText() {
		return String.format("+%s reputation with %s", gain, getLink());
	}
	
	@Override
	public String toString() {
		return String.format("+%s %s", gain, getLink());
	}
}

File: src/model/quest/ReputationGain.js
export default class ReputationGain {
	/**
	 * 
	 * @param {string} faction 
	 * @param {number} canonicalName 
	 * @param {number} gain
	 */
	constructor(faction, canonicalName, gain) {
		this.faction = faction;
		this.canonicalName = canonicalName;
		this.gain = gain;
	}

	/**
	 * @return {string}
	 */
	getLink() {
		if (this.canonicalName == null) {
			return "[[" + this.faction + "]]";
		} else {
			return "[[" + this.canonicalName + "|" + this.faction + "]]";
		}
	}

	/**
	 * @return {string}
	 */
	getFullText() {
		return "+" + this.gain + " reputation with " + this.getLink();
	}

	/**
	 * @return {string}
	 */
	toString() {
		return "+" + this.gain + " " + this.getLink();
	}
}
